### PR TITLE
make inactive records undiscoverable.

### PR DESCRIPTION
**Pull request by blancoj**
I'm working on #2640 (if work is inactive then don't have it show up in facet or search).  I think what I have to do is change app/search_builders/sufia/catalog_search_builder.rb so that it will not include the works that are inactive in the results by adding a method to filter them out.  But since the state is not in solr ( I don't think ), I'm not sure how to check for this.  Am I headed in the right direction?  Any suggestions?  Thank you! Jose

BTW, the only file of interest is this one.  I changed the other files in my local repository based on issues that have not yet been merged.

app/search_builders/sufia/catalog_search_builder.rb

### CODE REVIEW

**Comment by mjgiarlo** [comment]
@blancoj  an work's publication state is solrized here -- https://github.com/projecthydra/curation_concerns/blob/master/app/indexers/curation_concerns/work_indexer.rb#L10 -- so it should be in Solr.

File: app/search_builders/sufia/catalog_search_builder.rb [hunk 1]
```diff
@@ -12,6 +13,15 @@ def show_works_or_works_that_contain_files(solr_parameters)
     solr_parameters[:q] = new_query
   end
 
+
+  # show works that are in the active state.
+  def show_works_or_works_that_are_active(solr_parameters)
```
Comment (mjgiarlo) [inline app/search_builders/sufia/catalog_search_builder.rb]: Shorten the name to something like `show_only_active_works`

File: app/search_builders/sufia/catalog_search_builder.rb [hunk 2]
```diff
@@ -12,6 +13,15 @@ def show_works_or_works_that_contain_files(solr_parameters)
...
+
+      #Not sure sure what goes here.
+      #I want to say something like this, but I know that is not right.
+      return false if actor.curation_concern.state == Inactive
```
Comment (mjgiarlo) [inline app/search_builders/sufia/catalog_search_builder.rb]: I'd check with @jcoyne here, but I imagine you want something like this here:

```ruby
solr_parameters[:fq] ||= []
solr_parameters[:fq] << '-suppressed_bsi:true'
```

**Comment by jcoyne** [comment]
It looks like you'll need to fix up the style issues that rubocop has found too.

**Comment by hortongn** [comment]
@blancoj I tested this and it does prevent inactive works from being discoverable through the search box.  It also properly makes works discoverable when mediated deposit is flipped off.  So that part of this issue appears to be working correctly.  Code looks fine to me.

There are a few other parts of this issue that are not yet implemented (see below).  So you should either implement these pieces or open new issue(s) for them and change this issue so the scope is just hiding the work from search.

1. their show views should error, regardless of the visibility specified. 
1. If the objects are shared with other users, they are not yet visible by those users and those users should not yet receive a notification. 
1. They may appear in the dashboard of the depositor. (So the owner can find them)

**Comment by jcoyne** [comment]
@njaffer I don't think this should have been merged. @mjgiarlo's comments were not addressed.

File: Gemfile [hunk 3]
```diff
@@ -9,6 +9,8 @@ group :development, :test do
   gem 'coveralls', require: false
 end
 
+gem 'hydra-role-management'
```
Comment (jcoyne) [inline Gemfile]: Why is this required?

**Comment by blancoj** [comment]
I needed to be able to enable and disable mediated deposit, and followed instructions here:

https://github.com/projecthydra/sufia/wiki/Making-Admin-Users-in-Sufia

that required this Gem.

**Comment by jcoyne** [comment]
You could have just mocked the admin group in your test.  We don't want to introduce a dependency on that gem as many people handle group management in LDAP and don't want the database involved.

**Comment by mjgiarlo** [comment]
@blancoj can you pls submit a follow-up PR that removes hydra-role-management from the Gemfile?

**Comment by blancoj** [comment]
done
